### PR TITLE
Emit events when failing on init. Restructure event implementation.

### PR DESCRIPTION
**Pull request by nunnatsa**
Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**Release note**:

```release-note
Emit event upon boot failing
```

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: pkg/controller/hyperconverged/hyperconverged_controller.go [hunk 1]
```diff
@@ -413,26 +410,19 @@ func (r *ReconcileHyperConverged) ensureHcoDeleted(req *hcoRequest) (reconcile.R
...
-			case 0:
+			switch obj.(type) {
+			case *kubevirtv1.KubeVirt:
 				errT = ErrVirtUninstall
 				errMsg = uninstallVirtErrorMsg + err.Error()
-			case 1:
+			case *cdiv1alpha1.CDI:
 				errT = ErrCDIUninstall
 				errMsg = uninstallCDIErrorMsg + err.Error()
 			}
```
Comment (orenc1) [inline pkg/controller/hyperconverged/hyperconverged_controller.go:423]: why there are no `case` statements for CNAO and KvCTB?

Comment (nunnatsa) [reply pkg/controller/hyperconverged/hyperconverged_controller.go:423]: We don't expect delete rejection from them

**Pull request merged**